components/admin_console/billing/billing_subscriptions/index.tsx

```diff
@@ -0,0 +1,156 @@
+import {getCloudSubscription, getCloudProducts, getCloudCustomer} from 'mattermost-redux/actions/cloud';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {PreferenceType} from 'mattermost-redux/types/preferences';
+
+import {pageVisited, trackEvent} from 'actions/telemetry_actions';
+import FormattedAdminHeader from 'components/widgets/admin_console/formatted_admin_header';
+import {getCloudContactUsLink, InquiryType, InquiryIssue} from 'selectors/cloud';
+import {GlobalState} from 'types/store';
+import {
+    Preferences,
+    CloudBanners,
+    TELEMETRY_CATEGORIES,
+} from 'utils/constants';
+import {isCustomerCardExpired} from 'utils/cloud_utils';
+import {getRemainingDaysFromFutureTimestamp} from 'utils/utils.jsx';
+
+import BillingSummary from '../billing_summary';
+import PlanDetails from '../plan_details';
+
+import {
+    contactSalesCard,
+    cancelSubscription,
+    infoBanner,
+    creditCardExpiredBanner,
+    paymentFailedBanner,
+} from './billing_subscriptions';
+
+import './billing_subscriptions.scss';
+
+const WARNING_THRESHOLD = 3;
+
+type Props = {
+};
+
+const BillingSubscriptions: React.FC<Props> = () => {
+    const dispatch = useDispatch<DispatchFunc>();
+    const store = useStore();
+    const userLimit = useSelector((state: GlobalState) => parseInt(getConfig(state).ExperimentalCloudUserLimit!, 10));
+    const analytics = useSelector((state: GlobalState) => state.entities.admin.analytics);
+    const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
+    const isCloud = useSelector((state: GlobalState) => getLicense(state).Cloud === 'true');
+    const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
+
+    const products = useSelector((state: GlobalState) => state.entities.cloud.products);
+    const isCardExpired = useSelector((state: GlobalState) => isCustomerCardExpired(state.entities.cloud.customer));
+    const getCategory = makeGetCategory();
+    const preferences = useSelector<GlobalState, PreferenceType[]>((state) => getCategory(state, Preferences.ADMIN_CLOUD_UPGRADE_PANEL));
+
+    const contactSalesLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales));
+    const cancelAccountLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales, InquiryIssue.CancelAccount));
+
+    const [showCreditCardBanner, setShowCreditCardBanner] = useState(true);
+
+    let isFreeTrial = false;
+    let daysLeftOnTrial = 0;
+    if (subscription && subscription.trial_end_at! > 0 && subscription.status! === 'Trialing') {
```

Comment:
Why the non-null assertion here `subscription.status!` ? If we are certain that `Status` is not null, then we have it not null in the type definition 

Same for `trial_end_at`, it is defined as not null in the Subscription type so not sure why the non-null assertion would apply here

```suggestion
    if (subscription?.trial_end_at > 0 && subscription?.status === 'Trialing') {
```